tests/test_parser.py

```diff
@@ -36,104 +38,167 @@ class TestFileHandling(object):
+        srccode = ('# define MACRO {placeholder}\n'
+                   '\n'
+                   '# ifdef MACRO\n'
+                   '    # define MACRO2 placeholder2\n'
+                   '# endif\n')
+        lines = self.parser.fix_bad_code(srccode, rep).split('\n')
 
+        assert lines[0] == '# define MACRO 1'
+        assert lines[3] == '    # define MACRO2 2'
 
-    def test_load_file(self):
+        lines[0] = '# define MACRO {placeholder}'
+        lines[3] = '    # define MACRO2 placeholder2'
+        compare_lines(lines, srccode.split('\n'))
 
-        path = os.path.join(self.h_dir, 'replace.h')
-        assert self.parser.load_file(path)
-        assert self.parser.files[path] is not None
-        assert self.parser.file_order == [path]
-        assert self.parser.init_opts['replace']['replace.h'] is None
-        assert self.parser.init_opts['files'] == ['replace.h']
+    def test_load_non_existing_file(self):
+        path = self.hdr_file_path('no.h')
+        with raises(IOError):
+            self.parser.read(path)
+        assert path not in self.parser.file_order
 
-    def test_load_file_and_replace(self):
+    def test_removing_c_comments(self):
+        srccode = open(self.hdr_file_path('c_comments.h')).read()
+        nocomment_srccode = self.parser.remove_comments(srccode)
 
-        path = os.path.join(self.h_dir, 'replace.h')
-        rep = {'{placeholder}': '1', 'placeholder2': '2'}
-        assert self.parser.load_file(path, rep)
+        with open(self.hdr_file_path('c_comments_removed.h'), 'rU') as f:
+            compare_lines(nocomment_srccode.split('\n'), f.readlines())
 
-        lines = self.parser.files[path].split('\n')
-        assert lines[3] == '# define MACRO 1'
-        assert lines[6] == '    # define MACRO2 2'
+    def test_removing_cpp_comments(self):
+        srccode = open(self.hdr_file_path('cpp_comments.h')).read()
+        nocomment_srccode = self.parser.remove_comments(srccode)
+        with open(self.hdr_file_path('cpp_comments_removed.h'), 'rU') as f:
+            compare_lines(nocomment_srccode.split('\n'), f.readlines())
+
+    def test_process_multiple_files(self):
+        parser = CParser()
+        clib_intf = parser.clib_intf
+        parser.read(self.hdr_file_path('multi_file1.h'))
+        parser.read(self.hdr_file_path('multi_file2.h'))
+        assert parser.file_order == [self.hdr_file_path('multi_file1.h'),
+                                     self.hdr_file_path('multi_file2.h')]
+        assert clib_intf['MACRO_A'].content == 'replaced_val_a'
+        assert clib_intf['MACRO_B'].content == 'base_val_b'
+
+        parser.reset_clib_intf()
+        assert parser.file_order == []
+        parser.read(self.hdr_file_path('multi_file3.h'))
+        assert parser.clib_intf['MACRO_A'].content == 'other_val_a'
+        assert 'MACRO_B' not in parser.clib_intf
+        assert clib_intf['MACRO_A'].content == 'replaced_val_a'
+
+    def test_caching(self):
```

Comment:
here use the tmpdir fixture from pytest (simply add the argument to the function and use tmpdir as any directory path). It avoids messing with tempfile and shutil.